frameworks/TPOT/exec.py

```diff
@@ -67,6 +70,18 @@ def run(dataset, config):
     y_test = dataset.test.y
     with Timer() as predict:
         predictions = tpot.predict(X_test)
+
+    def infer(path):
+        data = pd.read_parquet(path)
+        if is_classification:
+            try:
+                return tpot.predict_proba(data)
+            except RuntimeError:
+                return tpot.predict(data)
+        return tpot.predict(data)
+
+    inference_times = measure_inference_times(infer, dataset.inference_subsample_files)
```

Comment:
Any consideration of loading the raw file and doing the preprocessing done by AMLB to make it compatible with frameworks like TPOT live? This would properly compute the true inference time, as currently TPOT skips the preprocessing overhead since it is given pre-processed data, unlike frameworks such as AutoGluon and H2O that take in the raw data and have to pre-process it as part of the inference call.

Specifically TPOT is given the encoded version of the data, while ignoring the time it takes to encode it:

```
    X_train, X_test = dataset.train.X_enc, dataset.test.X_enc
    y_train, y_test = unsparsify(dataset.train.y_enc, dataset.test.y_enc)
```

Beyond ignoring the time it take to encode it, it further benefits from the (in many cases) smaller data file artifact that would be able to load from disk faster than the raw file which is larger.

Reply:
I had a look at this, but it seems quite convoluted to achieve this (as files are currently generated outside of the subprocess). Since TPOT is the only automl framework affected by this, and the remainder can work on raw parquet/arff/csv files, I think I will leave this for now and exclude `TPOT` from the comparison (an unfortunate choice to add it first to this PR).